distro/src/main/java/io/camunda/migrator/app/MigratorApp.java

```diff
@@ -104,23 +105,30 @@ protected static void validateArguments(String[] args) {
       }
     }
 
+    // Check if we have too many flags (not counting entity type parameters)
+    if (flagCount > MAX_FLAGS) {
+      throw new IllegalArgumentException("Error: Too many arguments.");
+    }
+
+    // Validate skipped migration configuration
+    if (hasListSkipped && hasRetrySkipped) {
+      throw new IllegalArgumentException(
+          "Conflicting flags: --list-skipped and --retry-skipped cannot be used together. Please specify only one migration mode.");
```

Comment:
Nit: we could move this to above the for each loop to avoid having to do the loop if the combination is already invalid